src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/assist/HistoryCompletionManager.java

```diff
@@ -279,12 +363,21 @@ public String getHistory()
          return history_;
       }
       
+      public int getIndex()
+      {
+         return index_;
+      }
+      
       private final String history_;
       private final String match_;
+      private final int index_;
    }
    
    private CompletionListPopupPanel<HistoryMatch> popup_;
    private PopupMode mode_;
+   private RpcObjectList<HistoryEntry> lastResults_;
+   private String lastSearch_;
```

Comment:
Maybe initialize this to the empty string so we don't have to worry about whether it might be null or not?

Reply:
Good idea; done.